api/v1alpha1/instancegroup_types.go

```diff
@@ -42,6 +42,14 @@ const (
 	ReconcileErr   ReconcileState = "Error"
 )
 
+// InstanceGroupSpec defines the schema of resource Spec
+type InstanceGroupSpec struct {
+	Provisioner        string             `json:"provisioner"`
+	EKSCFSpec          *EKSCFSpec         `json:"eks-cf,omitempty"`
+	EKSFargateSpec     *EKSFargateSpec    `json:"eks-fargate,omitempty"`
+	AwsUpgradeStrategy AwsUpgradeStrategy `json:"strategy"`
+}
+
 // InstanceGroup is the Schema for the instancegroups API
 // +kubebuilder:object:root=true
 // +kubebuilder:resource:path=instancegroups,scope=Namespaced,shortName=ig
```

Comment:
These annotations need to remain above `InstanceGroupSpec`

Reply:
ok